### PR TITLE
Refactor/138 reorganize css

### PR DESCRIPTION
**Pull request by Andrew0914**
Hello 🤚

I worked on certain scss files to decouple some elements from their containers, specifically:

* I have decoupled the classes of `color_bar_component`,` color_bar_legend` and `table_component` from their containers, every class is independent now, but they still keep a relationship at the semantic level, I tried to follow the BEM standard a bit, but keeping the structure that you have, finally left the boostrap classes independent of the custom ones

For example:

```scss
.color-bar {
  &-progress {
    border-radius: 2px;
    ...
    &-title {
      color: $color-gray-700;
      ....
    }
}
```
The produced classes are: `color-bar-progress` and `color-bar-progress-title`

Also:

* resource_usage.scss was decomposed into: color_bar.scss and color_bar_legend which correspond to the components color_bar_component.ex and color_bar_legend.ex respectively.

* I renamed tabular_page.scss to table.scss to match the table_component, but had to prefix the class with `phx` to avoid overwriting the boostrap one.

* I moved the styles that only overwrite the bootstrap ones to the `extends` folder

* I moved some styles that are utilities or helpers for the content of the pages to the folder `utils`

### CODE REVIEW

File: lib/phoenix/live_dashboard/pages/os_mon_page.ex [hunk 1]
```diff
@@ -187,11 +187,11 @@ defmodule Phoenix.LiveDashboard.OSMonPage do
               <div class="card-body resource-usage">
```
Comment (bemesa21) [inline lib/phoenix/live_dashboard/pages/os_mon_page.ex]: I see that `resource-usage` and `resource-usage-total` were deleted in assets/css/app/components/_resource_usage.scss but here `resource-usage` is used.

Also in line `198` `resource-usage-total` is used.
Is it correct?

File: assets/css/app/components/_progress_section.scss [hunk 2]
```diff
@@ -1,14 +0,0 @@
-// Component with progress info and bar at the bottom (used for displaying resources usage etc.)
-.progress-section {
```
Comment (bemesa21) [inline assets/css/app/components/_progress_section.scss:2]: I found that `progress-section` is used in `lib/phoenix/live_dashboard/components/card_usage_component.ex` in `line number 6` and `progress` is used in `lib/phoenix/live_dashboard/components/title_bar_component.ex` as a child class of progress-section. I think that there could be a missing style. What do you think?

File: lib/phoenix/live_dashboard/pages/os_mon_page.ex (hunk 1 above)
Comment (bemesa21) [reply lib/phoenix/live_dashboard/pages/os_mon_page.ex]: also those classes are used in `lib/phoenix/live_dashboard/pages/home_page.ex`

File: assets/css/app/components/_progress_section.scss (hunk 2 above)
Comment (Andrew0914) [reply assets/css/app/components/_progress_section.scss:2]: Hi ✋🏽  , 

I forgot to remove those classes from the html, `resource-usage` and` progress-section` classes no longer exist and have no effect.

The class `resource-usage-total` and `resource-usage-total-value` do not belong to a component itself, but are used in 2 pages (`home` and` os_data`), so I moved  ` resource_usage.scss` with just those two classes to the utils folder.

Finally the `utils/resource_usage.scss`  file for some reason was empty, but I have already fixed it. 🔧 

@bemesa21

**Pull request closed without merging**